### ISSUE TITLE
ENHANCEMENT: `DropdownButton` Component

### ISSUE THREAD

**Issue by wijohnst** [wijohnst/thriving-park#161]
# Dropdown specific wrapper component around a`Button` component

![Image](https://user-images.githubusercontent.com/12022922/200205412-3bb2452f-3772-47fd-8f44-fc8a30bdbaa0.png)
_Screen shot, `Button` component with `CaretIcon`_

This `Button` is the default state and control for the `Dropdown` component. The final `Dropdown` component will have 2 children, `DropdownButton` and `DropdownMenu`:

```js
<Dropdown> // <- We will build this later
  <DropdownButton /> // <- This is what we are building
  <DropdownMenu /> // <- Will build this later
</Dropdown>
```

This pattern is modeled after the [`React Bootstrap` `Dropdown` component](https://react-bootstrap.netlify.app/components/dropdowns/#rb-docs-content)

**Comment by wijohnst** [comment]
# Separation of Concerns

[Separation of Concern](https://help.sap.com/doc/abapdocu_753_index_htm/7.53/en-US/abenseperation_concerns_guidl.htm#:~:text=Separation%20of%20concerns%20is%20a,and%20arrangement%20in%20software%20layers.) is the big thing that I'm starting to think about as we build the `DropdownButton` component. `DropdownButton` and `DropdownMenu` are both **[Dumb Components:](https://javascript.plainenglish.io/react-all-about-components-35650a02ff50#:~:text=Dumb%20/%20Stateless%20/%20Presentational%20Component)**

![Image](https://user-images.githubusercontent.com/12022922/200205862-455376ad-a647-4c73-8428-9bffaecbb590.png)
_"Dumb" Component Overview"_

Basically we need to make sure that most of the functionality for both of these components is passed down as props from either `Dropdown` (top-level component) or from some other parent context. Let's make sure we talk about his in detail.
